Various fixes

giig1967g opened the pull request:
%SP% = space
%ip4% = 4th IP octet

giig1967g committed:
Added 2 system variables

%SP% = space
%ip4% = 4th IP octet

micropet commented:
@giig1967g,
Do we have "free memory" or is this sysheap?

giig1967g commented:
in the code I see %sysheap% 
I can't see free memory.

TD-er commented:
I could have sworn ip1...ip4 were already present.

giig1967g commented:
ip1..ip4 are available as sysinfo output

Grovkillen commented:
No not as variables @TD-er . My source scan from a couple of weeks back didn't pick them up at least.

TD-er commented:
OK, strange, I thought it was already present.

giig1967g committed:
Merge remote-tracking branch 'upstream/mega' into mega

giig1967g committed:
Fixing bug #1142

giig1967g committed:
Revert #1170 that introduced a bug

The reason is that inside PLUGIN_WRITE the variable [event->BaseVarIndex] is not set and is always = 0.
So actually this PR modifies the 1st value of task 1 and not the encoder counter.
Line 183 of P059_Encoder.ino should be removed:
UserVar[event->BaseVarIndex] = (float) event->Par1;

My workaround so far is to add a taskvalueset just after the encwrite command.

giig1967g commented:
@TD-er : sorry I haven't been able to split the PRs.

TD-er commented:
That's OK, I will have a look at it tomorrow.

giig1967g commented:
Just for reference:
#1142: the bug was caused by the variable "haschanged" that was set to true in the init function, while it should have been "false"
#1170: The reason is that inside PLUGIN_WRITE you cannot read the taskID and the BaseVarIndex variables. They are always = 0. 
Line 183 of P059_Encoder.ino should be removed:
UserVar[event->BaseVarIndex] = (float) event->Par1;

giig1967g committed:
Merge remote-tracking branch 'upstream/mega' into mega

giig1967g committed:
Solution for #960

Added a new global command: TaskValueSetAndRun.

It changes the value of a task and runs it, so new value is pubished on MQTT and rules are executed.

giig1967g commented:
Sorry again TD-er: I really don't understand how can I submit different PR for every change I propose...

Grovkillen commented:
You should create a new branch for every suggested PR you want to submit.

giig1967g commented:
I am new to github   :(
so if I suggest 20 PR, I need 20 branches?
Isn't it a bit confusing?

Grovkillen commented:
No, not confusing at all. Better make small commits to keep them easier to review. The branches can be deleted with the click of a button once merged.

giig1967g commented:
I can try do redo the 4 PR with 4 branches.
Shall I do it?

Grovkillen commented:
Yes, it's good practice. :wink:

TD-er commented on src/Command.ino:
```diff
@@ -454,6 +455,19 @@ void ExecuteCommand(byte source, const char *Line)
+      float result = 0;
+      Calculate(TmpStr1, &result);
+      UserVar[(VARS_PER_TASK * (Par1 - 1)) + Par2 - 1] = result;
+      SensorSendTask(Par1 - 1);
```
Are you sure about all those "-1" ?

giig1967g replied:
Yes I am sure. The tasks commands (taskvalueset and taskrun) start from 1 while internally it starts from 0. Thus the minus 1.
My TaskValueSetAndRun is the merge of TaskValueSet + TaskRun.

giig1967g commented:
I have created 3 single PR.
Can close this one.
